fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -339,6 +349,7 @@ def no_sync(self) -> Generator:
         old_should_accumulate_grads = self.should_accumulate_grads
         self.should_accumulate_grads = True
         yield
+        self.accumulate_grads_flipped = self.should_accumulate_grads != old_should_accumulate_grads
```

Comment:
again, testing each and every combination exposed a broken case when we just went out of the no_sync context (the grads are not empty, but that's expected), which this fixes